### ISSUE TITLE
Stream#map fantasy land compliance / documentation

### ISSUE THREAD

**Issue by spacejack** [MithrilJS/mithril.js#2479]
**Mithril version:** 2.0.0-rc.9

Mithril's [implementation of Stream#map](https://github.com/MithrilJS/mithril.js/blob/next/stream/stream.js#L50) accepts an additional [undocumented](https://mithril.js.org/stream.html#streammap) parameter `ignoreInitial` which must have the value `stream.SKIP` to be recognized ([example](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvEAXwvW10QICsEqdBk2J4hcYgAI4ABkkBeSVnwSATjGwAKGQEoAOmnFS4ARgXSZ+LBgAOmgB4KAfJPsUlK4uuz4AygGkASQAFfTRZK1sHZ0lgcVpYfChaAHNNAHJZRHT3ex1WMNNIu0d5Fzj6OASYJNSM02zc-MKZTRMwyhA4GFhqYghKvABWRBk2DhBMHDx8ajgBGnpGZh42AF0qKAg0AGsEFE5pniwIYkJVaE6AV1VyHhJiGzhEAHoXq7QbHZTZ2iwXk5nC5QAACJnw4IAbADTudoPh+J1iABPGzcLrUC42UTsQ7owFwqAAWjUGiw11ueAeT1e70+31+-wJwJJXjJIIATPhLDIXqSfIiqCi0Xg4JiINj1qwgA).) I'm not sure if it's intended as part of the public API, but If it is we should add it to the docs.

However, it doesn't appear to be fully [compliant with the spec](https://mithril.js.org/stream.html#streamfantasy-landmap) which requires that it only take 1 parameter.

If this breaks the fantasy land contract then we should probably change it. It seems that the 2nd parameter is used only by `combine` internally. I would suggest we add a separate, public `dropInitial` function which could be used instead, and it would also be a handy feature for applications.

**Comment by isiahmeadows** [comment]
I wouldn't be against a quick edit to move that into a `._map` that `stream.map` and `Stream.combine` both delegate to. That'd end up looking something like this:

```js
stream._map = function(fn, ignoreInitial) {
	var target = ignoreInitial ? Stream(fn(value)) : Stream()
	target.parents.push(stream)
	dependentStreams.push(target)
	dependentFns.push(fn)
	return target
}

stream.map = function(fn) {
	return stream._map(fn, false)
}

// And for `Stream.combine`
function combine(fn, streams) {
	// ...
	var mappers = streams.map(function(s) {
		return s._map(function(value) {
			// ...
		}, s.state === "active")
	})
	// ...
}
```

**Comment by spacejack** [comment]
Would we document `_map`? If it's going to be publicly accessible shouldn't we make a more user-friendly version like `dropInitial(s: Stream, n: number)`?

**Comment by isiahmeadows** [comment]
Closed per #2481
